src/Magento/FunctionalTestingFramework/Util/Filesystem/CestFileCreatorUtil.php

```diff
@@ -0,0 +1,68 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\FunctionalTestingFramework\Util\Filesystem;
+
+use Magento\FunctionalTestingFramework\Exceptions\TestFrameworkException;
+
+class CestFileCreatorUtil
+{
+    /**
+     * Singleton CestFileCreatorUtil Instance.
+     *
+     * @var CestFileCreatorUtil
+     */
+    private static $INSTANCE;
+
+    /**
+     * CestFileCreatorUtil constructor.
+     */
+    private function __construct()
+    {
+    }
+
+    /**
+     * Get CestFileCreatorUtil instance.
+     *
+     * @return CestFileCreatorUtil
+     */
+    public static function getInstance(): CestFileCreatorUtil
+    {
+        if (!self::$INSTANCE) {
+            self::$INSTANCE = new CestFileCreatorUtil();
+        }
+
+        return self::$INSTANCE;
+    }
+
+    /**
+     * Create a single PHP file containing the $cestPhp using the $filename.
+     * If the _generated directory doesn't exist it will be created.
+     *
+     * @param string $filename
+     * @param string $exportDirectory
+     * @param string $testPhp
+     *
+     * @return void
+     * @throws TestFrameworkException
+     */
+    public function create(string $filename, string $exportDirectory, string $testPhp): void
```

Comment:
I like the idea of pulling some code out of TestGenerator, but I don't think that creating a singleton that hold no data for a single utility function makes sense.

I would sooner that they change the visibility of `createCestFile` to public and mock that method instead. This is a lot of code to introduce to mock a single method.

Reply:
@KevinBKozan, as for me, in general, the composition is much better than big classes with a variety of functions.
@jilu1, what do you think about that case?

Reply:
cc. @andrewbess

Reply:
@bohdan-harniuk Yeah, I see your point in that. I'm just vigilant about creating any classes that will perpetually live in memory, but I suppose that's not much of a concern if there's not a large amount of data being held.

I'm happy to defer to @jilu1 's original review in this case, I'll approve this and we can merge it.